### PR TITLE
Feature/prepaid balance management

### PR DESCRIPTION
**Pull request by Jason337**
This is the prepaid management stub.

### CODE REVIEW

File: app/controllers/prepay_balance_controller.js [hunk 1]
```diff
@@ -0,0 +1,1557 @@
+//  === Requirements ===
+const { query } = require('express')
```
Comment (SteanKotze) [inline app/controllers/prepay_balance_controller.js]: shouldn't be required

File: app/controllers/prepay_balance_controller.js [hunk 2]
```diff
@@ -0,0 +1,1557 @@
...
+                ]
+            }
+        ]
+    },
+    show_product_buckets: function(request)
+    {
+        console.log("prepay_balance_controller.show_product_buckets")
+
+        return {
+            id: faker.random.uuid(),
```
Comment (SteanKotze) [inline app/controllers/prepay_balance_controller.js]: please use the id being passed with the request when available (show, update, and deletes)

File: app/controllers/prepay_balance_controller.js [hunk 3]
```diff
@@ -0,0 +1,1557 @@
...
+                ]
+            }
+        ]
+    },
+    show_product_adjustments: function(request)
+    {
+        console.log("prepay_balance_controller.show_product_adjustments")
+
+        return {
+                routeDetails: {
```
Comment (SteanKotze) [inline app/controllers/prepay_balance_controller.js]: no id?

File: app/controllers/prepay_balance_controller.js [hunk 4]
```diff
@@ -0,0 +1,1557 @@
...
+                }
+        }
+    },
+    index_product_transfers: function(request)
+    {
+        console.log("prepay_balance_controller.index_product_transfers")
+
+        return [
+            {
+                routeDetails: {
```
Comment (SteanKotze) [inline app/controllers/prepay_balance_controller.js]: missing id

File: app/controllers/prepay_balance_controller.js [hunk 5]
```diff
@@ -0,0 +1,1557 @@
...
+                ]
+            }
+        ]
+    },
+    create_product_topups: function(request)
+    {
+        console.log("prepay_balance_controller.create_product_topups")
+
+        return {
+            routeDetails: {
```
Comment (SteanKotze) [inline app/controllers/prepay_balance_controller.js]: missing id

File: app/controllers/prepay_balance_controller.js [hunk 6]
```diff
@@ -0,0 +1,1557 @@
...
+    {
+        console.log("prepay_balance_controller.show_buckets")
+
+        return {
+
+            routeDetails: {
+                controller: 'Prepay Balance',
+                action: 'show_buckets'
+            },
+            id: faker.random.uuid(),
```
Comment (SteanKotze) [inline app/controllers/prepay_balance_controller.js]: wrong id

File: app/controllers/prepay_balance_controller.js [hunk 7]
```diff
@@ -0,0 +1,1557 @@
...
+                ]
+            }
+        ]
+    },
+    show_accumulated_balance_products: function(request)
+    {
+        console.log("prepay_balance_controller.show_accumulated_balance_products")
+
+        return {
+                routeDetails: {
```
Comment (SteanKotze) [inline app/controllers/prepay_balance_controller.js:695]: missing id

**Review by SteanKotze — CHANGES_REQUESTED** [review]

File: app/controllers/prepay_balance_controller.js (hunk 1 above)
Comment (Jason337) [reply app/controllers/prepay_balance_controller.js]: Removed

File: app/controllers/prepay_balance_controller.js (hunk 2 above)
Comment (Jason337) [reply app/controllers/prepay_balance_controller.js]: send back the bucket id

File: app/controllers/prepay_balance_controller.js (hunk 3 above)
Comment (Jason337) [reply app/controllers/prepay_balance_controller.js]: It is there(lower down), moved up for visibility

File: app/controllers/prepay_balance_controller.js (hunk 4 above)
Comment (Jason337) [reply app/controllers/prepay_balance_controller.js]: It is there(lower down), moved up for visibility

File: app/controllers/prepay_balance_controller.js (hunk 5 above)
Comment (Jason337) [reply app/controllers/prepay_balance_controller.js]: create does not require an ID

File: app/controllers/prepay_balance_controller.js (hunk 6 above)
Comment (Jason337) [reply app/controllers/prepay_balance_controller.js]: fixed

File: app/controllers/prepay_balance_controller.js (hunk 7 above)
Comment (Jason337) [reply app/controllers/prepay_balance_controller.js:695]: It is lower down under product id, I did fix the information though

**Review by SteanKotze — APPROVED** [review]

**Pull request merged**